### PR TITLE
Trim the JSON source in indexing slow logs

### LINKED ISSUE #38080: Plaintext slow logs with multiline causing problems when parsing
In order to keep the parsing logic consistent across docker and standard deployments, Beats is distinguishing the type of logs (JSON or plaintext) by the first character in a line.
If the first character is '{' then the JSON logging will be applied, otherwise the plaintext one.

Slow logs contain source field which wraps the json. When the `index.indexing.slowlog.reformat: false` is set, it will print the JSON as the user provided. Could be that the first line being `\n`. 
The plaintext log will end up like this:
```
[2018-07-04T21:51:30,411][INFO ][index.indexing.slowlog.index] [v_VJhjV] [metricbeat-6.3.0-2018.07.04/VLKxBLvUSYuIMKzpacGjRg] took[1.7ms], took_millis[1], type[doc], id[s01HZ2QBk9jw4gtgaFtn], routing[], source[
{
  "@timestamp":"2018-07-04T21:27:30.730Z",
  "metricset":{
    "name":"network",
    "module":"system",
    "rtt":7264},
    "system":{
      "network":{
        "name":"lo0",
        "in":{
          "errors":0,
          "dropped":0,
          "bytes":77666873,
          "packets":244595},
          "out":{
            "packets":244595,
            "bytes":77666873,
            "errors":0,
            "dropped":0
          }
        }
      },
      "beat":{
        "name":"Rados-MacBook-Pro.local",
        "hostname":"Rados-MacBook-Pro.local",
        "version":"6.3.0"
      },
      "host":{
        "name":"Rados-MacBook-Pro.local"
      }
    }]
```

This is causing problems for beats, as previously mentioned the first character in a line being `{` means it will try parse it as a JSON

The easiet approach will be to remove the `\n` at the beginning so that the first `{` ends up on the previous line
```
[2018-07-04T21:51:30,411][INFO ][index.indexing.slowlog.index] [v_VJhjV] [metricbeat-6.3.0-2018.07.04/VLKxBLvUSYuIMKzpacGjRg] took[1.7ms], took_millis[1], type[doc], id[s01HZ2QBk9jw4gtgaFtn], routing[], source[{
  "@timestamp":"2018-07-04T21:27:30.730Z",
...
...
}
```

The other approach would be to escape the JSON, but that would make it leas readable. Even less readable in `*.json` logs (escaping twice)

We should also schedule work to extract these fields into top level JSON fields in `*.json` logs 

the original discussion on beats PR
https://github.com/elastic/beats/pull/10447/files#r252512969

### PR DESCRIPTION
**Pull request by pgomulka**
The '{' as a first character in log line is causing problems for beats when parsing plaintext logs. 
This is used to distinguish if the logs are in JSON or plaintext (they want to keep the logic consistent for docker and standalone deployments, so file extension can not be used).

When the JSON in source element is **not** reformatted (`index.indexing.slowlog.reformat: false`) and the user send a document with an additional `\n` like here:
```
curl --request PUT \
  --url http://localhost:9200/index1/doc/2 \
  --header 'content-type: application/json' \
  --data '
{
      "network":{
        "name":"lo0",
        "in":{
          "errors":0,
          "dropped":0,
          "bytes":77666873,
          "packets":244595},
          "out":{
            "packets":244595,
            "bytes":77666873,
            "errors":0,
            "dropped":0
          }
        }
      }'
```
Then the `{` will endup in a new line
```
[2018-07-04T21:51:30,411][INFO ][index.indexing.slowlog.index] [v_VJhjV] [metricbeat-6.3.0-2018.07.04/VLKxBLvUSYuIMKzpacGjRg] took[1.7ms], took_millis[1], type[doc], id[s01HZ2QBk9jw4gtgaFtn], routing[], source[
{
  "@timestamp":"2018-07-04T21:27:30.730Z",
  "metricset":{
...
```
**Trimming the source part of a SlowLog will solve the problem and will keep the logs readable.**
```
2018-07-04T21:51:30,411][INFO ][index.indexing.slowlog.index] [v_VJhjV] [metricbeat-6.3.0-2018.07.04/VLKxBLvUSYuIMKzpacGjRg] took[1.7ms], took_millis[1], type[doc], id[s01HZ2QBk9jw4gtgaFtn], routing[], source[{
  "@timestamp":"2018-07-04T21:27:30.730Z",
...
...
}]
```
This won't happen for Search slow logs, as we repackage the user's request into our own `SearchRequest` object.
closes #38080

### CODE REVIEW

**Review by alpar-t — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/index/IndexingSlowLog.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ public String toString() {
             }
             try {
                 String source = XContentHelper.convertToJson(doc.source(), reformat, doc.getXContentType());
-                sb.append(", source[").append(Strings.cleanTruncate(source, maxSourceCharsToLog)).append("]");
+                sb.append(", source[").append(Strings.cleanTruncate(source, maxSourceCharsToLog).trim()).append("]");
```
Comment (danielmitterdorfer) [inline server/src/main/java/org/elasticsearch/index/IndexingSlowLog.java]: I wonder whether it is possible to create `source` without any leading line breaks so we can avoid the `trim()` at all?

**Review by danielmitterdorfer — COMMENTED** [review]
I left a comment. Based on the PR description I also could not grasp the purpose of this. It only became apparent after reading the related issue. Could you please reword the PR description as well so this becomes clearer?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 2]
```diff
@@ -261,16 +278,16 @@ private void assertTimeValueException(final IllegalArgumentException e, final St
...
     private IndexMetaData newIndexMeta(String name, Settings indexSettings) {
         Settings build = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
-            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
-            .put(indexSettings)
-            .build();
+                                 .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+                                 .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                                 .put(indexSettings)
+                                 .build();
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks a bit odd to me and also unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 3]
```diff
@@ -261,16 +278,16 @@ private void assertTimeValueException(final IllegalArgumentException e, final St
         assertThat(e.getCause(), instanceOf(IllegalArgumentException.class));
         final IllegalArgumentException cause = (IllegalArgumentException) e.getCause();
         final String causeExpected =
-                "failed to parse setting [" + key + "] with value [NOT A TIME VALUE] as a time value: unit is missing or unrecognized";
+            "failed to parse setting [" + key + "] with value [NOT A TIME VALUE] as a time value: unit is missing or unrecognized";
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 4]
```diff
@@ -247,7 +264,7 @@ public void testSetLevels() {
         try {
             settings.updateIndexMetaData(newIndexMeta("index",
                 Settings.builder().put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_WARN_SETTING.getKey(), "NOT A TIME VALUE")
-                    .build()));
+                        .build()));
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 5]
```diff
@@ -238,7 +255,7 @@ public void testSetLevels() {
         try {
             settings.updateIndexMetaData(newIndexMeta("index",
                 Settings.builder().put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_INFO_SETTING.getKey(), "NOT A TIME VALUE")
-                    .build()));
+                        .build()));
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 6]
```diff
@@ -229,7 +246,7 @@ public void testSetLevels() {
         try {
             settings.updateIndexMetaData(newIndexMeta("index",
                 Settings.builder().put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_DEBUG_SETTING.getKey(), "NOT A TIME VALUE")
-                    .build()));
+                        .build()));
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 7]
```diff
@@ -220,7 +237,7 @@ public void testSetLevels() {
         try {
             settings.updateIndexMetaData(newIndexMeta("index",
                 Settings.builder().put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_TRACE_SETTING.getKey(), "NOT A TIME VALUE")
-                    .build()));
+                        .build()));
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 8]
```diff
@@ -202,8 +219,8 @@ public void testSetLevels() {
         assertEquals(TimeValue.timeValueMillis(420).nanos(), log.getIndexWarnThreshold());
 
         metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .build());
+                                                 .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 9]
```diff
@@ -191,9 +208,9 @@ public void testSetLevels() {
 
         settings.updateIndexMetaData(newIndexMeta("index",
             Settings.builder().put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_TRACE_SETTING.getKey(), "120ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_DEBUG_SETTING.getKey(), "220ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_INFO_SETTING.getKey(), "320ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_WARN_SETTING.getKey(), "420ms").build()));
+                    .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_DEBUG_SETTING.getKey(), "220ms")
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 10]
```diff
@@ -176,12 +193,12 @@ public void testLevelSetting() {
 
     public void testSetLevels() {
         IndexMetaData metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_TRACE_SETTING.getKey(), "100ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_DEBUG_SETTING.getKey(), "200ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_INFO_SETTING.getKey(), "300ms")
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_THRESHOLD_INDEX_WARN_SETTING.getKey(), "400ms")
-            .build());
+                                                               .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 11]
```diff
@@ -154,8 +171,8 @@ public void testLevelSetting() {
         assertEquals(SlowLogLevel.TRACE, log.getLevel());
 
         metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .build());
+                                                 .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 12]
```diff
@@ -127,12 +128,28 @@ public void testReformatSetting() {
...
+        assertThat(logLine.split("\n").length, equalTo(1));
+    }
+
     public void testLevelSetting() {
         SlowLogLevel level = randomFrom(SlowLogLevel.values());
         IndexMetaData metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_LEVEL_SETTING.getKey(), level)
-            .build());
+                                                               .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 13]
```diff
@@ -106,8 +107,8 @@ public void testReformatSetting() {
         assertTrue(log.isReformat());
 
         metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .build());
+                                                 .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 14]
```diff
@@ -82,15 +83,15 @@ public void testSlowLogParsedDocumentPrinterSourceToLog() throws IOException {
...
+            + " was expecting ('true', 'false' or 'null')\n"
+            + " at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper")));
     }
 
     public void testReformatSetting() {
         IndexMetaData metaData = newIndexMeta("index", Settings.builder()
-            .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
-            .put(IndexingSlowLog.INDEX_INDEXING_SLOWLOG_REFORMAT_SETTING.getKey(), false)
-            .build());
+                                                               .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 15]
```diff
@@ -82,15 +83,15 @@ public void testSlowLogParsedDocumentPrinterSourceToLog() throws IOException {
         assertNotNull(e.getCause());
         assertThat(e.getCause(), instanceOf(JsonParseException.class));
         assertThat(e.getCause(), hasToString(containsString("Unrecognized token 'invalid':"
-                + " was expecting ('true', 'false' or 'null')\n"
-                + " at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper")));
+            + " was expecting ('true', 'false' or 'null')\n"
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 16]
```diff
@@ -46,10 +47,10 @@
 public class IndexingSlowLogTests extends ESTestCase {
     public void testSlowLogParsedDocumentPrinterSourceToLog() throws IOException {
         BytesReference source = BytesReference.bytes(JsonXContent.contentBuilder()
-            .startObject().field("foo", "bar").endObject());
+                                                                 .startObject().field("foo", "bar").endObject());
         ParsedDocument pd = new ParsedDocument(new NumericDocValuesField("version", 1),
             SeqNoFieldMapper.SequenceIDFields.emptySeqID(), "id",
-                "test", null, null, source, XContentType.JSON, null);
+            "test", null, null, source, XContentType.JSON, null);
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java [hunk 17]
```diff
@@ -46,10 +47,10 @@
 public class IndexingSlowLogTests extends ESTestCase {
     public void testSlowLogParsedDocumentPrinterSourceToLog() throws IOException {
         BytesReference source = BytesReference.bytes(JsonXContent.contentBuilder()
-            .startObject().field("foo", "bar").endObject());
+                                                                 .startObject().field("foo", "bar").endObject());
```
Comment (danielmitterdorfer) [inline server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: That looks unrelated to this change? Can you please restore the old formatting?

**Review by danielmitterdorfer — COMMENTED** [review]
The changes themselves look fine but I think we should not reformat unrelated changes here so I left a bunch of comments around that.

File: server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java (hunk 2 above)
Comment (pgomulka) [reply server/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java]: I reverted back to previous formatting. must have done this by accident. sorry

**Review by danielmitterdorfer — APPROVED** [review]
Thanks for iterating. LGTM
